### ISSUE TITLE
all: true option does not merge interfaces

### ISSUE THREAD

**Issue by terion-name** [okgrow/merge-graphql-schemas#141]
```js
import { importSchema } from 'graphql-import'
const schema = mergeTypes([
      importSchema("schema/application-schema.graphql"),
      importSchema("schema/admin-schema.graphql"),
    ], {all: true});
```

`importSchema` allows to import types from other schemas for nice code organization:
```graphql
# import Article, ArticleWhereInput, ArticleOrderByInput, ArticleCreateInput from "./generated-schema.graphql"
type Query {
    articles(
        where: ArticleWhereInput
        orderBy: ArticleOrderByInput
        skip: Int
        after: String
        before: String
        first: Int
        last: Int
    ): [Article!]!
}
```

When this imports are in multiple files - they duplicate and during merge I need to eliminate doubles.

`{all: true}` handles this for types. But interfaces are still doubling

**Comment by renato** [comment]
I'm trying it with Prisma too and, in addition to interface types, Prisma also defines scalar types (like DateTime) which also get imported more than once.

I added `Kind.INTERFACE_TYPE_DEFINITION` and `Kind.SCALAR_TYPE_DEFINITION` on

https://github.com/okgrow/merge-graphql-schemas/blob/b79816c63574eb899120ea435e24825d2100035a/src/utilities/astHelpers.js#L6-L9

and used `nodes = []` on

https://github.com/okgrow/merge-graphql-schemas/blob/b79816c63574eb899120ea435e24825d2100035a/src/merge_types.js#L17

and all tests still pass.

I couldn't find another way of using merge with import so I'm using this workaround for now.

The `Kind.*_TYPE_DEFINITION` not checked there are `ENUM`, `OPERATION` and `UNION`, I don't know if they could eventually be imported twice (and so "defined more than once").

**Comment by heresandyboy** [comment]
Hi,
I am also stuck with this issue, with a combination of Prisma, merge-graphql-schemas and graphql-import. Using the flag { all: true} with mergeTypes. The merge results in Error: Type "Node" was defined more than once. 

I have just been trying to modularise the advanced-typescript Prisma boilerplate, to hand to my team as an example of scaling the application. 
I really don't want to have to resort to copy-pasting all of the types, cascading down for types/interfaces/scalars that are already generated by Prisma.

Has anyone found another way to achieve merging with Prisma type imports?

**Comment by renato** [comment]
@heresandyboy I'm not using `merge-graphql-schemas` currently, I replaced it by manually glob'ing & reading files from a pattern (e.g. `**/schema.graphql`), joining them and passing to `graphql-import`'s `importSchema` function (just notice the relative path of the `# import` directives will be different). An ugly workaround but the only way I could get it to work.

Something like...
```js
const typeDefsImports = glob
  .sync(path.join(__dirname, "**/schema.graphql"))
  .map(file => fs.readFileSync(file, "utf-8"))

const typeDefs = importSchema(typeDefsImports.join("\n"))
```

At first I tried creating a parent `schema.graphql` but I couldn't get the nested importing to work (`parent-schema.graphql` importing from `product/schema.graphql` which imports from the `generated-prisma.graphql`).
